openapi/tests/openapi_integration/test_multicollection_reco.py

```diff
@@ -0,0 +1,174 @@
+        method="POST",
+        path_params={'collection_name': collection_name},
+        body={
+            "limit": 3,
+            "positive": [1],
+            "with_vector": False,
+            "with_payload": True,
+            "lookup_from": {
+                "collection": collection_name2,
+                "vector": "other"
+            }
+        }
+    )
+    assert response.ok, response.text
+    assert len(response.json()['result']) == 3
+    assert response.json()['result'][0]['payload'] is not None
+
+    response = request_with_validation(
+        api='/collections/{collection_name}/points/recommend/batch',
+        method="POST",
+        path_params={'collection_name': collection_name},
+        body={
+            "searches": [
+                {
+                    "limit": 3,
+                    "positive": [1],
+                    "with_vector": False,
+                    "with_payload": True,
+                    "lookup_from": {
+                        "collection": collection_name2,
+                        "vector": "other"
+                    }
+                },
+                {
+                    "limit": 3,
+                    "positive": ["00000000-0000-0000-0000-000000000000"],
+                    "with_vector": False,
+                    "with_payload": True,
+                    "lookup_from": {
+                        "collection": collection_name2,
+                        "vector": "other"
+                    }
+                },
+                {
+                    "limit": 3,
+                    "positive": [1],
+                    "with_vector": False,
+                    "with_payload": True,
+                }
+            ]
+        }
+    )
+    assert response.ok, response.text
+    assert len(response.json()['result']) == 3
+    assert len(response.json()['result'][0]) == 3
+    assert len(response.json()['result'][1]) == 3
+    assert len(response.json()['result'][2]) == 3
+    assert response.json()['result'][0][0]['payload'] is not None
+
+    response = request_with_validation(
+        api='/collections/{collection_name}/points/recommend',
+        method="POST",
+        path_params={'collection_name': collection_name},
+        body={
+            "limit": 3,
+            "positive": [1],
+            "lookup_from": {
+                "collection": "unknown_collection",
+                "vector": "other"
+            }
+        }
+    )
+    assert response.status_code == 404
+
+    response = request_with_validation(
+        api='/collections/{collection_name}/points/recommend',
+        method="POST",
+        path_params={'collection_name': collection_name},
+        body={
+            "limit": 3,
+            "positive": [1],
+            "lookup_from": {
+                "collection": collection_name2,
+                "vector": "unknown_vector"
+            }
+        }
+    )
+    assert response.status_code == 400, response.text
```

Comment:
why are returning a 400 here and not a 404?

Reply:
originally we had 400 if there is no vector with given name, if you think we need to change that, I propose to do it in another PR

Reply:
I think it would be more consistent to have a 404 but it is not a strong opinion.
Tackling this in a different PR is of course fine :+1: